web_src/js/features/common-global.js

```diff
@@ -79,6 +79,33 @@ function doRedirect(redirect) {
   form.submit();
 }
 
+async function fetchActionDoRequest(actionElem, url, opt) {
+  try {
+    const resp = await fetch(url, opt);
+    if (resp.status === 200) {
+      let {redirect} = await resp.json();
+      redirect = redirect || actionElem.getAttribute('data-redirect');
+      actionElem.classList.remove('dirty'); // remove the areYouSure check before reloading
+      if (redirect) {
+        fetchActionDoRedirect(redirect);
+      } else {
+        window.location.reload();
+      }
```

Comment:
I know this is old code, but shouldn't we use 302 status code and then check [resp.redirected](https://developer.mozilla.org/en-US/docs/Web/API/Response/redirected) and read location from [resp.url](https://developer.mozilla.org/en-US/docs/Web/API/Response/url)? I think this is something for a future  generic fetch handler.

Reply:
Nope IMO.

Fetch's `redirect` could only have 3 behaviors: follow, error, manual.

`follow` is the default behavior, it doesn't distinguish 200/302, etc, `error` just stops redirection.

If you would like to use `"redirect": "manual"` .... IIRC it doesn't work well (the details are too complex)

You could take a try, if it could succeed, feel free to propose a new solution.

Reply:
Yes, that was implying `manual` option, but I guess it'll be hard to distinguish a "content redirect" from an "ui redirect", so I guess I'm fine with leaving it unchanged.